staging/src/k8s.io/component-base/metrics/legacyregistry/registry.go

```diff
@@ -43,7 +43,7 @@ func init() {
 // Deprecated: Please note the issues described in the doc comment of
 // InstrumentHandler. You might want to consider using promhttp.Handler instead.
 func Handler() http.Handler {
-	return prometheus.InstrumentHandler("prometheus", promhttp.HandlerFor(defaultRegistry, promhttp.HandlerOpts{}))
+	return promhttp.InstrumentMetricHandler(prometheus.DefaultRegisterer, promhttp.HandlerFor(defaultRegistry, promhttp.HandlerOpts{}))
```

Comment:
Why are we doing this?

Reply:
prometheus 1.0.0 dropped the InstrumentHandler method

Reply:
https://github.com/kubernetes/kubernetes/pull/83987/files#diff-fb2a0310eaebea082d9732696d1e4e02

Reply:
I changed this part because otherwise it conflict updated prometheous

Reply:
This will end up being backwards incompatible, since the metrics exposed are actually different.

Reply:
what's the backwards compatible way to emulate this registration?

Reply:
Literally copying over the old handler code. I don't think it's worth it though.

While this may break some people's instrumented setups, it breaks a more meta-level of monitoring (scrape frequency and metric endpoint request stuff) and is less critical than control-plane health metrics. 

My preference is to keep @wenjiaswe's changes and have an accompanying release note on the removal of the old prometheus meta-metrics.

Reply:
> My preference is to keep @wenjiaswe's changes and have an accompanying release note on the removal of the old prometheus meta-metrics.

Ok, I thought you were saying we could not make this change

Reply:
@logicalhan do you have a suggested release note in mind or could you check what I have below? Not sure how to write release note for breaking changes. Do we include in the PR note or is there a separate place? 

```
Use [promhttp.InstrumentMetricHandler](https://godoc.org/github.com/prometheus/client_golang/prometheus/promhttp#InstrumentMetricHandler) instead of [prometheus.InstrumentHandler](https://godoc.org/github.com/dnesting/client_golang/prometheus#InstrumentHandler) for metrics handler, please refer to the linked doc for the difference.
```

Reply:
> Ok, I thought you were saying we could not make this change

That is what I was initially thinking, but after looking into the actual metrics this would end up breaking, it seems both low risk/value. 

> do you have a suggested release note in mind

Since we're removing metrics, maybe it would be a good idea to list them? Something like this maybe?
```
Deprecated prometheus request meta-metrics have been removed (http_request_duration_microseconds, http_request_duration_microseconds_sum, http_request_duration_microseconds_count, http_request_size_bytes, http_request_size_bytes_sum, http_request_size_bytes_count, http_requests_total, http_response_size_bytes, http_response_size_bytes_sum, http_response_size_bytes_count) due to removal from the prometheus client library. Prometheus http request meta-metrics are now generated from [promhttp.InstrumentMetricHandler](https://godoc.org/github.com/prometheus/client_golang/prometheus/promhttp#InstrumentMetricHandler) instead.
```

Reply:
Thanks! Updated in release note.